snorkel/contrib/fonduer/fonduer/models/context.py

```diff
@@ -521,3 +548,88 @@ def __ne__(self, other):
 
     def __hash__(self):
         return id(self)
+
+
+class TemporaryImage(TemporaryContext):
+    """The TemporaryContext version of Figure"""
+    def __init__(self, figure):
+        super(TemporaryImage, self).__init__()
+        self.figure   = figure  # The figure Context
+
+    def __len__(self):
+        return 1
+
+    def __eq__(self, other):
+        try:
+            return self.figure.position == other.figure.position \
+                and self.figure.url == other.figure.url
+        except AttributeError:
+            return False
+
+    def __ne__(self, other):
+        try:
+            return self.figure.position != other.figure.position \
+                or self.figure.url != other.figure.url
+        except AttributeError:
+            return True
+
+    def __contains__(self, other_span):
+        return self.__eq__(other_span)
+
+    def __hash__(self):
+        return hash(self.figure)
+
+    def get_stable_id(self):
+        return "%s::%s:%s" % (self.figure.document.id, self._get_polymorphic_identity(), self.figure.position)
+
+    def _get_table_name(self):
+        return 'image'
+
+    def _get_polymorphic_identity(self):
+        return 'image'
+
+    def _get_insert_query(self):
+        return """INSERT INTO image VALUES(:id, :document_id, :image_id, :url)"""
+
+    def _get_insert_args(self):
+        return {'document_id': self.figure.document.id,
+                'image_id' : self.figure.position,
+                'url'      : self.figure.url}
+
+    def __repr__(self):
+        return '%s(document=%s, position=%s, url=%s)' \
+            % (self.__class__.__name__, self.figure.document.name.encode('utf-8'), \
+                self.figure.position, self.figure.url)
+
+    def _get_instance(self, **kwargs):
+        return TemporaryImage(**kwargs)
+
+
+class Image(Context, TemporaryImage):
+    """
+    A span of characters, identified by Context id and character-index start, end (inclusive).
```

Comment:
An image is a "span of characters"? 

We should clarify the comments so that it's clear how this is different that just a Span?